example_configs/spark.yml

```diff
@@ -12,39 +12,70 @@ rules:
     name: spark_driver_$2_$3
     labels:
       app_id: "$1"
 
+  # These come from the application driver
+  # Emulate histograms for CodeGenerator
+  - pattern: "metrics<name=(.*)\\.driver\\.CodeGenerator\\.(.*)><>(Count|Min|Max|(\\d+)thPercentile|Mean|StdDev)"
+    name: spark_driver_CodeGenerator_$2_$3
+    labels:
+      app_id: "$1"
+
+  # These come from the application driver
+  # Emulate histograms for LiveListenerBus
+  - pattern: "metrics<name=(.*)\\.driver\\.LiveListenerBus\\.(.*)><>(Count|Min|Max|(\\d+)thPercentile|Mean|StdDev|DurationUnit|RateUnit|MeanRate|OneMinuteRate|FifteenMinuteRate)"
+    name: spark_driver_LiveListenerBus_$2_$3
+    labels:
+      app_id: "$1"
+
   # These come from the application driver if it's a streaming application
   # Example: app-20160809000059-0000.driver.com.example.ClassName.StreamingMetrics.streaming.lastCompletedBatch_schedulingDelay
   - pattern: "metrics<name=(.*)\\.driver\\.(.*)\\.StreamingMetrics\\.streaming\\.(.*)><>Value"
-    name: spark_streaming_driver_$3
+    name: spark_driver_streaming_$3
     labels:
       app_id: "$1"
       app_name: "$2"
 
   # These come from the application driver if it's a structured streaming application
   # Example: app-20160809000059-0000.driver.spark.streaming.QueryName.inputRate-total
   - pattern: "metrics<name=(.*)\\.driver\\.spark\\.streaming\\.(.*)\\.(.*)><>Value"
-    name: spark_structured_streaming_driver_$3
+    name: spark_driver_structured_streaming_$3
     labels:
       app_id: "$1"
       query_name: "$2"
 
   # These come from the application executors
-  # Example: app-20160809000059-0000.0.executor.threadpool.activeTasks
-  - pattern: "metrics<name=(.*)\\.(.*)\\.executor\\.(.*)><>Value"
+  # Example: app-20160809000059-0000.0.executor.threadpool.activeTasks (value)
+  #  app-20160809000059-0000.0.executor.JvmGCtime (counter)
+  - pattern: "metrics<name=(.*)\\.(.*)\\.executor\\.(.*)><>(Value|Count)"
     name: spark_executor_$3
     labels:
       app_id: "$1"
       executor_id: "$2"
 
-  # These come from the master
-  # Example: application.com.example.ClassName.1470700859054.cores
-  - pattern: "metrics<name=application\\.(.*)\\.([0-9]+)\\.(.*)><>Value"
-    name: spark_application_$3
+
+  # These come from the application executors
+  # Example: app-20160809000059-0000.0.jvm.threadpool.activeTasks
+  - pattern: "metrics<name=(.*)\\.([0-9]+)\\.(jvm|HiveExternalCatalog|NettyBlockTransfer)\\.(.*)><>(Count|Value)"
+    name: spark_executor_$3_$4
+    labels:
+      app_id: "$1"
+      executor_id: "$2"
+
+
+  # These come from the application driver
+  # Emulate histograms
+  - pattern: "metrics<name=(.*)\\.([0-9]+)\\.CodeGenerator\\.(.*)><>(Count|Min|Max|(\\d+)thPercentile|Mean|StdDev)"
+    name: spark_executor_CodeGenerator_$3_$4
     labels:
-      app_name: "$1"
-      app_start_epoch: "$2"
+      app_id: "$1"
+      executor_id: "$2"
+
+scrape_configs:
```

Comment:
I don't think you meant to put this here

Reply:
Ok I thought it was useful. It is just an example, but yes can be removed.